internal/batches/service/service.go

```diff
@@ -55,32 +57,41 @@ const sourcegraphVersionQuery = `query SourcegraphVersion {
 }
 `
 
-// getSourcegraphVersion queries the Sourcegraph GraphQL API to get the
-// current version of the Sourcegraph instance.
-func (svc *Service) getSourcegraphVersion(ctx context.Context) (string, error) {
+// getSourcegraphVersionAndMaxChangesetsCount queries the Sourcegraph GraphQL API to get the
+// current version and max unlicensed changesets count for the Sourcegraph instance.
+
+func (svc *Service) getSourcegraphVersionAndMaxChangesetsCount(ctx context.Context) (string, int, error) {
 	var result struct {
-		Site struct {
+		MaxUnlicensedChangesets int
+		Site                    struct {
 			ProductVersion string
 		}
 	}
 
-	ok, err := svc.client.NewQuery(sourcegraphVersionQuery).Do(ctx, &result)
+	ok, err := svc.client.NewQuery(getInstanceInfo).Do(ctx, &result)
 	if err != nil || !ok {
-		return "", err
+		return "", 0, err
 	}
 
-	return result.Site.ProductVersion, err
+	return result.Site.ProductVersion, result.MaxUnlicensedChangesets, err
 }
 
-// DetermineFeatureFlags fetches the version of the configured Sourcegraph and
-// returns the enabled features.
-func (svc *Service) DetermineFeatureFlags(ctx context.Context) (*batches.FeatureFlags, error) {
-	version, err := svc.getSourcegraphVersion(ctx)
+// DetermineFeatureFlagsAndLicense returns the enabled features and license restrictions
+// configured for the Sourcegraph instance.
+
+func (svc *Service) DetermineFeatureFlagsAndLicense(ctx context.Context) (*batches.LicenseRestrictions, *batches.FeatureFlags, error) {
+	version, mc, err := svc.getSourcegraphVersionAndMaxChangesetsCount(ctx)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to query Sourcegraph version to check for available features")
+		return nil, nil, errors.Wrap(err, "failed to query Sourcegraph version and license info for instance capabilities")
```

Comment:
```suggestion
		return nil, nil, errors.Wrap(err, "failed to query Sourcegraph version and license info for instance")
```

I think this reads better. Just a suggestion.